### DISCUSSION TITLE
GPUでジョブを実行する際のエラー：no CUDA-capable device is detected について

### DISCUSSION THREAD

**Discussion by marimasunaga** [nig-sc/nigsc_homepage2#377] [Q&A] [answered]
いつも大変お世話になっております。

CuPyを使った計算を試みています。
ジョブを投入して実行しようとすると、
cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorNoDevice: no CUDA-capable device is detected
とエラーが出ます。

ちなみにGPUのログインノード(qlogin -l gpu -l cuda=1でログイン)でpythonのコードを実行するとちゃんと計算できています。

以下がジョブ投入の際のスクリプトですが、不備が無いかご確認いただけないでしょうか。
```
#!/bin/bash

#$ -cwd 
#$ -V 
#$ -l gpu
#$ -l cuda=1     
#$ -l d_rt=10:00:00
#$ -l s_rt=10:00:00
#$ -l s_vmem=12G 
#$ -l mem_req=12G
#$ -N cupy
#$ -S /bin/bash

export CUDA_HOME=/cm/shared/apps/cuda11.2/toolkit/11.2.0
export PATH=$CUDA_HOME/bin:$PATH
export LD_LIBRARY_PATH=$CUDA_HOME/lib64:$LD_LIBRARY_PATH

pip install cupy
nvcc --version
nvidia-smi

input_path="/home/mari1618033/3D/gpu.py"
python3 "$input_path"
```

こちらを
`qsub -l gpu -l cuda=1 cupy.sh`
として投入しております。

Outputファイルでは、
CuPyに関してはRequirement already satisfied、
nvcc —versionとnvidia -smiは正しく出力できている状況です。

知識不足で大変恐縮ですが、何卒よろしくお願いいたします。

**Comment by sc-fs** [comment, marked as answer]
スクリプトを修正して頂きたいのですが、
・「export CUDA_VISIBLE_DEVICES="0"」の追加gpu0を利用する定義の追加
・以下は必要ないのと思われるので削除

```
#$ -V 
export CUDA_HOME=/cm/shared/apps/cuda11.2/toolkit/11.2.0
export PATH=$CUDA_HOME/bin:$PATH
export LD_LIBRARY_PATH=$CUDA_HOME/lib64:$LD_LIBRARY_PATH

pip install cupy
```


以下のような形のスクルプトをqsub cupy.shしてしていただけませんか。
(cupy.sh))
```
#!/bin/bash

#$ -cwd 
#$ -l gpu
#$ -l cuda=1     
#$ -l d_rt=10:00:00
#$ -l s_rt=10:00:00
#$ -l s_vmem=12G 
#$ -l mem_req=12G
#$ -N cupy
#$ -S /bin/bash

export CUDA_VISIBLE_DEVICES="0"

nvcc --version
nvidia-smi

input_path="/home/mari1618033/3D/gpu.py"
python3 "$input_path"
```



よろしくお願いします。

**Reply by marimasunaga** [reply]
ご回答どうもありがとうございます。
無事に計算できていました。
大変助かりました。
